sdk/python/kfp/components/component_decorator_test.py

```diff
@@ -106,3 +107,46 @@ def comp(text: str) -> str:
         component_spec = structures.ComponentSpec.load_from_component_yaml(
             yaml_text)
         self.assertEqual(component_spec, comp.component_spec)
+
+    def test_output_named_tuple_with_dict(self):
+
+        @component
+        def comp(
+                text: str) -> NamedTuple('outputs', [('data', Dict[str, str])]):
+            outputs = NamedTuple('outputs', [('data', Dict[str, str])])
+            return outputs(data={text: text})
+
+        # TODO: ideally should be the canonical type string, rather than the specific annotation as string, but both work
+        self.assertEqual(comp.component_spec.outputs['data'].type,
+                         'typing.Dict[str, str]')
+
+    def test_output_dict(self):
+
+        @component
+        def comp(text: str) -> Dict[str, str]:
+            return {text: text}
+
+        # TODO: ideally should be the canonical type string, rather than the specific annotation as string, but both work
+        self.assertEqual(comp.component_spec.outputs['Output'].type,
```

Comment:
NIT: Should it be `comp.component_spec.output` instead of `comp.component_spec.outputs['Output']`?

Reply:
In this case it should not be.

This is different than the data passing syntax used within a pipeline definition. This is checking the in-memory representation of the component, wherein all outputs (`OutputSpec`s) are included in the dictionary `.outputs`, irrespective of how many outputs there are.

Reply:
Got it, thank you for the explanation!